Code/CryGame/Items/Weapons/WeaponSystem.cpp

```diff
@@ -456,6 +463,142 @@ int  CWeaponSystem::QueryProjectiles(SProjectileQuery& q)
     return q.nCount;
 }
 
+//------------------------------------------------------------------------
+void CWeaponSystem::Scan(const char* folderName)
+{
+	string folder = folderName;
+	string search = folder;
+	search += "/*.*";
+
+	ICryPak* pPak = m_pSystem->GetIPak();
+
+	_finddata_t fd;
+	intptr_t handle = pPak->FindFirst(search.c_str(), &fd);
+
+	if (!m_recursing)
+		CryLog("Loading ammo XML definitions from '%s'!", folderName);
+
+	if (handle > -1)
+	{
+		do
+		{
+			if (!strcmp(fd.name, ".") || !strcmp(fd.name, ".."))
+				continue;
+
+			if (fd.attrib & _A_SUBDIR)
+			{
+				string subName = folder + "/" + fd.name;
+				if (m_recursing)
+					Scan(subName.c_str());
+				else
+				{
+					m_recursing = true;
+					Scan(subName.c_str());
+					m_recursing = false;
+				}
+				continue;
+			}
+
+			if (_stricmp(CryPath::GetExt(fd.name), "xml"))
+				continue;
+
+			string xmlFile = folder + string("/") + string(fd.name);
+			XmlNodeRef rootNode = m_pSystem->LoadXmlFile(xmlFile.c_str());
+
+			if (!rootNode)
+			{
+				CryLogWarningAlways("Invalid XML file '%s'! Skipping...", xmlFile.c_str());
+				continue;
+			}
+
+			if (!ScanXML(rootNode, xmlFile.c_str()))
+				continue;
+
+		} while (pPak->FindNext(handle, &fd) >= 0);
+	}
```

Comment:
The original code from Mod SDK leaks CryPak search handles due to missing `ICryPak::FindClose` call. Better to use our own fancy `CryFind`:

```suggestion
	if (!m_recursing)
		CryLog("Loading ammo XML definitions from '%s'!", folderName);

	for (auto& entry : CryFind(search.c_str()))
	{
		if (entry.IsDirectory())
		{
			string subName = folder + "/" + entry.name;
			if (m_recursing)
			{
				Scan(subName.c_str());
			}
			else
			{
				m_recursing=true;
				Scan(subName.c_str());
				m_recursing=false;
			}
			continue;
		}

		if (_stricmp(CryPath::GetExt(entry.name), "xml"))
		{
			continue;
		}

		string xmlFile = folder + string("/") + string(entry.name);
		XmlNodeRef rootNode = m_pSystem->LoadXmlFile(xmlFile.c_str());

		if (!rootNode)
		{
			CryLogWarning("Invalid XML file '%s'! Skipping...", xmlFile.c_str());
			continue;
		}

		ScanXML(rootNode, xmlFile.c_str());
	}
```